packages/perseus-core/src/parse-perseus-json/perseus-parsers/hint.ts

```diff
@@ -1,11 +1,17 @@
-import {any, boolean, object, string} from "../general-purpose-parsers";
+import {
+    any,
+    boolean,
+    object,
+    optional,
+    string,
+} from "../general-purpose-parsers";
 import {defaulted} from "../general-purpose-parsers/defaulted";
 
 import {parseImages} from "./images-map";
 import {parseWidgetsMap} from "./widgets-map";
 
 export const parseHint = object({
-    replace: defaulted(boolean, () => undefined),
+    replace: defaulted(optional(boolean), () => undefined),
```

Comment:
I'm not clear why you needed to wrap `optional()` here? Is it because the `replace` can be `undefined` and if we didn't wrap it in `optional()` here, we'd have a resulting type of `boolean` because the fallback is ignored for type inference?

Reply:
Yep!